Update documentation for Theming to use sx props (#7331)
marmelab/react-admin: docs/Theming.md

## Overriding A Component Style

> Every react-admin component inherits an `sx` property from MUI, which is always applied to the root element.

Reviewer: ```suggestion
Every react-admin component exposes the `sx` property from MUI, which is always applied to the root element.
```

Reviewer 2: The `className` information is still valid. I wouldn't replace one over the other, but add the `sx` prop to it

```jsx
import * as React from 'react';
import { NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';

> const ColoredNumberFieldClasses = {

Reviewer: ```suggestion
const ColoredNumberFieldStyles = {
```

    small: { color: 'black' },
    big: { color: 'red' },
};

const ColoredNumberField = (props) => (
    <NumberField
        sx={{
            ...(props.record[props.source] < 100 &&

>                 ColoredNumberFieldClasses.small),

Reviewer: ```suggestion
                ColoredNumberFieldStyles.small),
```

            ...(props.record[props.source] >= 100 &&

>                 ColoredNumberFieldClasses.big),

Reviewer: ```suggestion
                ColoredNumberFieldStyles.big),
```

```jsx
import * as React from 'react';
import { NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';

const ColoredNumberFieldClasses = {
    small: { color: 'black' },
    big: { color: 'red' },
};

const colored = (WrappedComponent) => (props) =>
    (
        <WrappedComponent
            sx={{
                ...(props.record[props.source] < 100 &&

>                     ColoredNumberFieldClasses.small),

Reviewer: ```suggestion
                    ColoredNumberFieldStyles.small),
```

                ...(props.record[props.source] >= 100 &&

>                     ColoredNumberFieldClasses.big),

Reviewer: ```suggestion
                    ColoredNumberFieldStyles.big),
```
